pkg/validate/print.go

```diff
@@ -22,29 +19,49 @@ func getTableHeaders() *simpletable.Header {
 			{Align: simpletable.AlignCenter, Text: "FAILURE MESSAGE"},
 		},
 	}
+
+	table.SetStyle(simpletable.StyleCompactLite)
+
+	return table
 }
 
-func newSuccessTableRow(v types.Validator) []*simpletable.Cell {
-	return newTableRow(v, statusSuccess, "")
+type resultTable struct {
+	*simpletable.Table
 }
 
-func newFailedTableRow(v types.Validator, failureMsg string) []*simpletable.Cell {
-	return newTableRow(v, statusFailed, failureMsg)
+func (t resultTable) WriteRow(row []*simpletable.Cell) {
+	t.Body.Cells = append(t.Body.Cells, row)
 }
 
-func newErrorTableRow(v types.Validator, err error) []*simpletable.Cell {
-	return newTableRow(v, statusError, err.Error())
+func (t resultTable) WriteResult(res types.ValidatorResult) {
+	row := resultToRow(res)
+
+	if res.IsSuccess() {
```

Comment:
I think this refactor has it's purpose. But the side effect is we are going over the result logic thrice now:
1. inside `WriteResult`
2. inside `resultToRow`
3. inside `ValidateCLI`

Can we go over the result logic only once?

Reply:
Uh, in a sense yes. I played around with this by having a `ResultList` which abstracts some details away https://github.com/Ajpantuso/addon-metadata-operator/blob/8bbc42b81b54a2959e8b076c55b191800b5e8c44/internal/cmd/validate.go#L102-L124. But, if you mean have a single loop where everything gets done I'm not a fan. Three loops that do one thing each is more readable and just as efficient as one loop that does three things. And if you use those `List` abstractions you can kind of make the loops disappear into the background.